### ISSUE TITLE
Neither Python 3.13.0b nor Python 3.14 works with Pillow

### ISSUE THREAD

**Issue by xqt** [python-pillow/Pillow#8080]
### What did you do?
`pip install "Pillow!=10.0,!=10.1,>=8.1.2"`

### What did you expect to happen?
Pillow should be installed for Python 3.13 as well as Python 3.14

### What actually happened?
Python 3.13:
--
https://github.com/wikimedia/pywikibot/actions/runs/9236742053/job/25412925104
```
Downloading packaging-24.0-py3-none-any.whl (53 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.5/53.5 kB 13.3 MB/s eta 0:00:00
Building wheels for collected packages: pytest-attrib, coverage
  Building wheel for pytest-attrib (pyproject.toml): started
  Building wheel for pytest-attrib (pyproject.toml): finished with status 'done'
  Created wheel for pytest-attrib: filename=pytest_attrib-0.1.3-py3-none-any.whl size=4651 sha256=6a76e29006433b16c7c506d4877dce562e7a2fd35791346f1d62b5696d48afd0
  Stored in directory: /home/runner/.cache/pip/wheels/32/6b/92/f94dd527023ca74ec9189e79c3cbbcd2c8aae7dbd4ad886f8c
  Building wheel for coverage (pyproject.toml): started
  Building wheel for coverage (pyproject.toml): finished with status 'done'
  Created wheel for coverage: filename=coverage-7.5.1-cp313-cp313-linux_x86_64.whl size=236289 sha256=939e15b4efcff7b8ece5887929c140b81437dafc008a7e3bfd29dee0dbd0307e
  Stored in directory: /home/runner/.cache/pip/wheels/36/5b/3d/79b0b9f8983ad9aff6708469cfce66c13248cd23f639aeda3e
Successfully built pytest-attrib coverage
Installing collected packages: snowballstemmer, fake-useragent, pyflakes, pydocstyle, pycodestyle, pluggy, packaging, mccabe, isort, iniconfig, coverage, attrs, pytest, flake8, verve-flake8-mock, pytest-subtests, pytest-cov, pytest-attrib, flake8-isort, flake8-docstrings
Successfully installed attrs-23.2.0 coverage-7.5.1 fake-useragent-1.5.1 flake8-7.0.0 flake8-docstrings-1.7.0 flake8-isort-6.1.1 iniconfig-2.0.0 isort-5.13.2 mccabe-0.7.0 packaging-24.0 pluggy-1.5.0 pycodestyle-2.11.1 pydocstyle-6.3.0 pyflakes-3.2.0 pytest-8.2.1 pytest-attrib-0.1.3 pytest-cov-5.0.0 pytest-subtests-0.12.1 snowballstemmer-2.2.0 verve-flake8-mock-0.4
Ignoring importlib_metadata: markers 'python_version < "3.8"' don't match your environment
Collecting mwparserfromhell>=0.5.2 (from -r requirements.txt (line 22))
  Downloading mwparserfromhell-0.6.6.tar.gz (138 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 138.9/138.9 kB 6.7 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: packaging in /opt/hostedtoolcache/Python/3.13.0-beta.1/x64/lib/python3.13/site-packages (from -r requirements.txt (line 23)) (24.0)
Collecting requests>=2.21.0 (from -r requirements.txt (line 24))
  Downloading requests-2.32.2-py3-none-any.whl.metadata (4.6 kB)
Collecting mwoauth!=0.3.1,>=0.2.4 (from -r requirements.txt (line 34))
  Downloading mwoauth-0.4.0-py3-none-any.whl.metadata (2.1 kB)
Collecting pydot>=1.4.1 (from -r requirements.txt (line 37))
  Downloading pydot-2.0.0-py3-none-any.whl.metadata (9.6 kB)
Collecting python-stdnum>=1.19 (from -r requirements.txt (line 40))
  Downloading python_stdnum-1.20-py2.py3-none-any.whl.metadata (18 kB)
Collecting Pillow!=10.0,!=10.1,>=8.1.2 (from -r requirements.txt (line 43))
  Downloading pillow-10.3.0.tar.gz (46.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.6/46.6 MB 63.0 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/Python/3.13.0-beta.1/x64/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
          ~~~~^^
        File "/opt/hostedtoolcache/Python/3.13.0-beta.1/x64/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ~~~~^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.13.0-beta.1/x64/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line [118](https://github.com/wikimedia/pywikibot/actions/runs/9236742053/job/25412925104#step:5:119), in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-87agmo37/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-87agmo37/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
          ~~~~~~~~~~~~~~^^
        File "/tmp/pip-build-env-87agmo37/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
          ~~~~^^^^^^^^^^^^^^^^
        File "<string>", line 33, in <module>
        File "<string>", line 27, in get_version
      KeyError: '__version__'
      [end of output]
```
Python 3.14
--
https://github.com/wikimedia/pywikibot/actions/runs/9236742053/job/25412925198
```
Collecting Pillow!=10.0,!=10.1,>=8.1.2 (from -r requirements.txt (line 43))
  Downloading pillow-10.3.0.tar.gz (46.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.6/46.6 MB 67.4 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      Traceback (most recent call last):
        File "/home/runner/venv-3.14/lib/python3.14/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
          ~~~~^^
        File "/home/runner/venv-3.14/lib/python3.14/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ~~~~^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/runner/venv-3.14/lib/python3.14/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line [118](https://github.com/wikimedia/pywikibot/actions/runs/9236742053/job/25412925198#step:5:119), in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-2bs9qsl8/overlay/lib/python3.14/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-2bs9qsl8/overlay/lib/python3.14/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
          ~~~~~~~~~~~~~~^^
        File "/tmp/pip-build-env-2bs9qsl8/overlay/lib/python3.14/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
          ~~~~^^^^^^^^^^^^^^^^
        File "<string>", line 33, in <module>
        File "<string>", line 27, in get_version
      KeyError: '__version__'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
```
### What are your OS, Python and Pillow versions?

* OS: Ubuntu 22.04.4
* Python: 3.13.0b1, 3.14.0a0
* Pillow: 10.3.0

**Comment by Yay295** [comment]
see #8075

**Comment by hugovk** [comment]
Thanks for early testing! Please install Pillow from source or wait until the next release in July.

**Comment by radarhere** [comment]
This error was fixed by #8050

**Comment by radarhere** [comment]
Pillow 10.4.0 has just been released. If you try again now, the problem should be resolved.
